### PR TITLE
[EHT-550] Feature/apt installation

### PR DESCRIPTION
**Pull request by kowshikvajrala**

### CODE REVIEW

File: roles/node/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  - "{{ node_log_dir }}"
+
+node_config_file_path: "{{ node_config_dir }}/{{ node_config_file_name }}"
+node_log_file_path: "{{ node_log_dir }}/{{ service_identifier }}.log"
+
+node_environment_general_file_path: "{{ node_env_dir }}/{{ node_environment_general_file_name }}"
+node_environment_secrets_file_path: "{{ node_env_dir }}/{{ node_environment_secrets_file_name }}"
+
+node_private_key_path: "{{ node_key_dir }}/{{ node_private_key_file_name }}"
+node_public_key_path: "{{ node_key_dir }}/{{ node_public_key_file_name }}"
```
Comment (aditya-manit) [inline roles/node/defaults/main.yml:25]: do we need this ?

File: roles/node/tasks/install_binary.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    path: "{{ item.path }}"
+    state: absent
+  with_items: "{{ mina_files.files }}"
+  when: mina_files.matched > 0 
+
+- name: Add mina protocol repository
+  ansible.builtin.deb822_repository:
+    name: "mina-{{ mina.validator.release_type }}"
+    types: deb
+    uris: http://packages.o1test.net
```
Comment (aditya-manit) [inline roles/node/tasks/install_binary.yml:19]: lets put it in source_url and use it here

File: roles/node/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  - "{{ node_home_dir }}"
+  - "{{ node_bin_dir }}"
+  - "{{ node_config_dir }}"
+  - "{{ node_env_dir }}"
+  - "{{ node_log_dir }}"
+
+node_config_file_path: "{{ node_config_dir }}/{{ node_config_file_name }}"
+node_log_file_path: "{{ node_log_dir }}/{{ service_identifier }}.log"
+
+node_environment_general_file_path: "{{ node_env_dir }}/{{ node_environment_general_file_name }}"
```
Comment (aditya-manit) [inline roles/node/defaults/main.yml:21]: node_environment_general_file_path is not being used

Comment (aditya-manit) [reply roles/node/defaults/main.yml:21]: I think you have moved everything to system service already

File: roles/node/templates/mainnet/systemd.service.j2 [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+Group={{ service_identifier }}
+WorkingDirectory={{ node_home_dir }}
+EnvironmentFile={{ node_environment_secrets_file_path }}
+
+ExecStart={{ node_bin_dir }}/{{ service_identifier }} daemon --config-directory {{ node_config_dir }} \
+  --peer-list-url "https://storage.googleapis.com/mina-seed-lists/mainnet_seeds.txt" \
+  --log-level Info --file-log-level Debug \
+  --external-port {{ mina.validator.ports.external_port }} \
+  --metrics-port {{ mina.validator.ports.metrics_port }} \
+  --block-producer-key {{ node_private_key_path }} \
```
Comment (aditya-manit) [inline roles/node/templates/mainnet/systemd.service.j2:15]: --block-producer-key is DEPRECATED use env variable

File: roles/node/templates/mainnet/systemd.service.j2 [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+[Service]
+User={{ service_identifier }}
+Group={{ service_identifier }}
+WorkingDirectory={{ node_home_dir }}
+EnvironmentFile={{ node_environment_secrets_file_path }}
+
+ExecStart={{ node_bin_dir }}/{{ service_identifier }} daemon --config-directory {{ node_config_dir }} \
+  --peer-list-url "https://storage.googleapis.com/mina-seed-lists/mainnet_seeds.txt" \
+  --log-level Info --file-log-level Debug \
```
Comment (aditya-manit) [inline roles/node/templates/mainnet/systemd.service.j2:12]: 2024-12-13 11:10:49 UTC [Error] libp2p_helper: failed to set log level debug for routing/record: error: No such logger


check this once, are both of them required ? if yes then set both to info, but I think only should be required

File: roles/node/templates/mainnet/systemd.service.j2 [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+Description={{ service_identifier }}
+
+[Service]
+User={{ service_identifier }}
+Group={{ service_identifier }}
+WorkingDirectory={{ node_home_dir }}
+EnvironmentFile={{ node_environment_secrets_file_path }}
+
+ExecStart={{ node_bin_dir }}/{{ service_identifier }} daemon --config-directory {{ node_config_dir }} \
+  --peer-list-url "https://storage.googleapis.com/mina-seed-lists/mainnet_seeds.txt" \
```
Comment (aditya-manit) [inline roles/node/templates/mainnet/systemd.service.j2:11]: lets fetch this from group vars

File: roles/node/templates/mainnet/systemd.service.j2 [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  --log-level Info --file-log-level Debug \
+  --external-port {{ mina.validator.ports.external_port }} \
+  --metrics-port {{ mina.validator.ports.metrics_port }} \
+  --block-producer-key {{ node_private_key_path }} \
+  --limited-graphql-port {{ mina.validator.ports.graphql_port }} \
+  --uptime-submitter-key {{ node_private_key_path }} \
+  --uptime-url "https://uptime-backend.minaprotocol.com/v1/submit"
```
Comment (aditya-manit) [inline roles/node/templates/mainnet/systemd.service.j2:18]: lets fetch this from group vars

File: roles/node/templates/testnet/systemd.service.j2 [hunk 8]
```diff
@@ -0,0 +1,30 @@
+[Unit]
+Description={{ service_identifier }}
+
+[Service]
+User={{ service_identifier }}
+Group={{ service_identifier }}
+WorkingDirectory={{ node_home_dir }}
+EnvironmentFile={{ node_environment_secrets_file_path }}
+
+ExecStart={{ node_bin_dir }}/{{ service_identifier }} daemon --config-directory {{ node_config_dir }} \
```
Comment (aditya-manit) [inline roles/node/templates/testnet/systemd.service.j2:10]: make the same changes here as well

File: group_vars/all.yml [hunk 9]
```diff
@@ -14,20 +14,9 @@ default_node_types:
...
-    source_url: https://releases.sui.io
-    ports:
-      server_listen_port: 9191
-      metrics_port: 9186
-  bridge:
-    source_url: https://releases.sui.io
-    ports:
-      server_listen_port: 9191
-      metrics_port: 9186
+      external_port: 8302
```
Comment (aditya-manit) [inline group_vars/all.yml:20]: do we need 8303 anywere too ?

File: tasks/create_user_and_directories.yml [hunk 10]
```diff
@@ -17,6 +17,14 @@
     mode: "0755"
   loop: "{{ directories }}"
 
+- name: Create mina key directory
+  ansible.builtin.file:
+    path: "{{ node_key_dir }}"
+    state: directory
+    owner: "{{ service_identifier }}"
+    group: "{{ service_identifier }}"
+    mode: "0700"
```
Comment (aditya-manit) [inline tasks/create_user_and_directories.yml:26]: does it not start without 700 ?

File: roles/node/templates/testnet/systemd.service.j2 [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+  --peer-list-url "https://storage.googleapis.com/mina-seed-lists/mainnet_seeds.txt" \
```
Comment (aditya-manit) [inline roles/node/templates/testnet/systemd.service.j2:11]: mainnet_seeds.txt in testnet ?

**Review by aditya-manit — CHANGES_REQUESTED** [review]

File: roles/node/templates/mainnet/systemd.service.j2 (hunk 4 above)
Comment (kowshikvajrala) [reply roles/node/templates/mainnet/systemd.service.j2:15]: "monitor.ml.Error",
    [ "Base58_check.Invalid_base58_character(\"Private key\")" ],
    
    when i use the env varibale it's throwing this error

File: roles/node/templates/mainnet/systemd.service.j2 (hunk 5 above)
Comment (kowshikvajrala) [reply roles/node/templates/mainnet/systemd.service.j2:12]: That error is present even if it's not set

**Review by aditya-manit — APPROVED** [review]

**Pull request merged**